File: life-program-institute/src/main/java/com/belphegor/lifeprograminstitute/dao/UserDAO.java
package com.belphegor.lifeprograminstitute.dao;

import com.belphegor.common.mybatis.BaseDAO;
import com.belphegor.lifeprograminstitute.dao.mapper.UserMapper;
import com.belphegor.lifeprograminstitute.entity.User;
import org.springframework.stereotype.Component;

@Component
public class UserDAO extends BaseDAO<User, UserMapper> {
}

File: manage/src/main/kotlin/com/web/common/dao/UserDao.kt
package com.web.common.dao

import com.web.common.entity.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
open class UserDao{

    @Autowired
    lateinit var userMapper: UserMapper

    fun findUserByUserId(userId:String): User {
        return userMapper.selectByUserId(userId)
    }

    fun insertSelective(user: User):Int{
        return userMapper.insertSelective(user)
    }

}